### PR TITLE
Pet evolution bug fixes

### LINKED ISSUE #2135: Pets egg bug
[x]: # (**********************************)
[x]: # (** Fill in the following fields **)
[x]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Issue Prelude

[//]: # (Please complete these mandatory steps and check the following boxes by putting an `x` inside the brackets _before_ filing your issue)

-  [x] I have not modified the source prior to reproducing this issue.
-  [x] I am using the latest version of Hercules.
-  [x] I am aware that this report will be closed or deleted if it becomes obvious that I am stating the false.

### Description
Pets are not dissapearing its own eggs after you hatch them using an incubator.

[//]: # (Description of the problem or issue at length.)
[//]: # (Please specify any battle configuration related to the components of this issue that have been changed from the default values. This will allow quicker determination of the cause of the problem.)

### Current Behavior

Any pet egg is working like this... they remain in the inventory as an unknown item... which is not normal. At the beginning the egg disappears, but if you do refresh command, the egg will show up in your inventory. If using a magnifier it will detect the egg as an unknown item.

[//]: # (Describe at length what you noticed during your analysis.)
[//]: # (If this is a crash, post the core/stack-dump or crash-log to https://gist.github.com/)
[//]: # (If you are referencing from sources such as iROwiki or ratemyserver.net, please quote specific information rather than providing the links alone.)

### Expected Behavior

The egg should disappear after hatching the pet, and it shouldn't come back in your inventory unless you return the pet to the egg... But in this case  is detecting the egg as an existing item that returns as soon as you walk through a warp or use @refresh command...

[//]: # (Tell us what should happen instead.)

### Steps To Reproduce The Issue

1. Catch a pet or Make an egg... for example: @item 9055
2. Hatch it using a pet incubator.
3. change the map, warp or @refresh... Pet egg will appear in your inventory as an Unknown item

**Branch(es):**
- [x] master
- [x] Stable

**Hercules rev. hash/commit:** 
Latest Stable and Master

### Operating System

CentOS
(Sorry i don't know how to fill the checklist, but i have not modified anything from the src)

### PR DESCRIPTION
**Pull request by Asheraf**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

this pull request fixes some bugs related to pet evolution

**Issues addressed:**

fixes #2135

[//]: # (Issue Tracker Number if any.)

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2136)

File: src/map/pet.c [hunk 1]
```diff
@@ -345,7 +345,7 @@ static int pet_return_egg(struct map_session_data *sd, struct pet_data *pd)
 			pd->pet.pet_id == MakeDWord(sd->status.inventory[i].card[1], sd->status.inventory[i].card[2]));
 
 	if (i != MAX_INVENTORY) {
-		sd->status.inventory[i].identify = 1;
+		sd->status.inventory[i].attribute ^= ATTR_BROKEN;
```
Comment (4144) [inline src/map/pet.c]: probably better unset attribute from broken state. to prevent situations if it was in wrong state

File: src/map/pet.c [hunk 2]
```diff
@@ -348,7 +348,10 @@ static int pet_return_egg(struct map_session_data *sd, struct pet_data *pd)
 		sd->status.inventory[i].attribute ^= ATTR_BROKEN;
 		sd->status.inventory[i].bound = IBT_NONE;
 	}
-
+#if PACKETVER >= 20170704
```
Comment (4144) [inline src/map/pet.c]: wrong year used. should be 2018 :)

File: src/map/pet.c [hunk 3]
```diff
@@ -462,6 +465,9 @@ static int pet_birth_process(struct map_session_data *sd, struct s_pet *petinfo)
 		clif->spawn(&sd->pd->bl);
 		clif->send_petdata(sd,sd->pd, 0,0);
 		clif->send_petdata(sd,sd->pd, 5,battle_config.pet_hair_style);
+#if PACKETVER >= 20170704
```
Comment (4144) [inline src/map/pet.c]: and here

File: src/map/clif.c [hunk 4]
```diff
@@ -7233,6 +7233,7 @@ static void clif_sendegg(struct map_session_data *sd)
 ///     3 = accessory
 ///     4 = performance (data = 1~3: normal, 4: special)
 ///     5 = hairstyle
+///     6 = close egg selection ui and update egg in inventory (PACKETVER >= 20170704)
```
Comment (4144) [inline src/map/clif.c]: should be 2018

**Comment by MishimaHaruna** [comment]
If there are pets in use, the inventory table needs to be updated for this, right? If so, a sql update script might be a good idea - it could update all the records where card0 has the special pet value and identify is 0

**Pull request merged**